hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/common/model/HoodieSparkRecord.java

```diff
@@ -257,26 +258,12 @@ public HoodieRecord truncateRecordKey(Schema recordSchema, Properties props, Str
   }
 
   @Override
-  protected boolean checkIsDelete(Schema recordSchema, Properties props) {
+  protected boolean checkIsDelete(Schema recordSchema, Properties props, DeleteContext deleteContext) {
```

Comment:
some thoughts:

- the param should be just `DeleteContext`, the context already holds the record schema;
- hoodie operation should be handled separately because the meta fields are not there in write path.
- for the other logic, go through `xxx.getFieldAccessorInstance().isDeleteRecord`, that is how we unify the logic or read/write paths.